routers/web/repo/commit.go

```diff
@@ -90,7 +90,7 @@ func Commits(ctx *context.Context) {
 	commitsTagsMap, err := repo_model.FindTagsByCommitIDs(ctx, ctx.Repo.Repository.ID, commitIDs...)
 	if err != nil {
 		log.Error("FindTagsByCommitIDs: %v", err)
-		ctx.Flash.Error(ctx.Tr("repo.commit.load_tags_failed"))
+		ctx.Flash.Error(ctx.Tr("load_failed", "tags"))
```

Comment:
Word tags won't be translated, also I don't think in such cases how to even translate things as I have already mentioned it previously in some languages each word can have multiple writings (similar like English have for verbs in current and past tense). For example in Latvian translating such sentence word "tags" would be different writing than used elsewhere where it's not part of such sentence

Reply:
It should be simplified to:

```
internal_error_skipped = Internal error occurred but skipped: %s
```

Then do not translated the internal message, just put a function name or special word there. Users could see:

```
Internal Error: FindTagsByCommitIDs
```

The reason is that there will be many different "internal errors", in most cases these skippable "internal errors" should never happen, showing the real function name would help reporting and debugging, and won't really bother end users.

Reply:
6dfaa720dbb38db88f3a29e49781157266da9258

Reply:
Imho in all cases error can just be "Internal error occurred while loading data" as we already have error in logs and from security perspective it's not recommended to show user technical error details. Also there is not much end user (not admin) can do with such information anyway

Reply:
> Imho in all cases error can just be "Internal error occurred while loading data" as we already have error in logs

No. If you spent time on helping users to debug problem, you would know that a general sentence doesn't help much. It is really helpful to show some function names or special words.

> and from security perspective it's not recommended to show user technical error details

No, a function name or a special word is not "user technical error details"

> Also there is not much end user (not admin) can do with such information anyway

No, even "admin" can not do much with such information.

A user or an admin could just "report" in most cases.

Reply:
@wxiaoguang I have been helping users to debug problems but I don't think gitea open source project should not be dealing with end users but admin users that can check logs for errors etc

Anyway I don't agree that end users should see technical details in UI but I won't be blocking this also

Reply:
That's just my opinion that `Internal error occurred but is skipped: FindTagsByCommitIDs` is better.

I won't block a common sentence like `Internal error occurred while loading data` either, so please @go-gitea/technical-oversight-committee make final decision.

Reply:
Disclosing the function name can be considered a CWE, for which CVEs have been issued. Advice from OWASP (and similar orgs) is: tldr: log the more verbose data incl func name/stack trace etc.., and present a more generic error to the user. "Internal Error: Failed to load tags" or something similar is what they'd recommend.

Reply:
But what's the difference? Since Gitea is an open-source project, there will be only one "Internal Error: Failed to load tags", it exactly matches that line in code.

Almost all error messages from Gitea exposes the line in code.

Reply:
ps: `Internal Error: Failed to load tags` or `Internal Error: FindTagsByCommitIDs` either looks to me, they all behave equivalently, even for security.

My major concern is that such error message doesn't need to be translated, otherwise, there would be a lot of duplicate (and meaningless) translations:

```
    "Load users failed because of internal error"
    "Load branches failed because of internal error"
    "Load workflows failed because of internal error"
    "Load releases failed because of internal error"
    "Load projects failed because of internal error"
    "Load labels failed because of internal error"
    "Load milestones failed because of internal error"
    ....
```

Reply:
a1091923ecd72788774ad76aa234ea30a2050d83 please review again.